src/deriver/ScopedKeys.js

```diff
@@ -0,0 +1,105 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const HKDF = require('node-hkdf');
+const base64url = require('base64url');
+
+const KEY_LENGTH = 32;
+
+/**
+ * Scoped key deriver
+ * @desc Used by the Firefox Accounts content server
+ * @module deriver-ScopedKeys
+ * @example
+ * ```js
+ * const scopedKeys = new fxaCryptoDeriver.ScopedKeys();
+ *
+ * return scopedKeys.deriveScopedKeys({
+ *   identifier: 'https://identity.mozilla.com/apps/notes',
+ *   inputKey: 'bc3851e9e610f631df94d7883d5defd5e5f55ab520bd5a9ae33dae26575c6b1a',
+ *   keyMaterial: '0000000000000000000000000000000000000000000000000000000000000000',
+ *   timestamp: 1494446722583
+ * });
+ * ```
+ */
+class ScopedKeys {
+  /**
+   * Derive a scoped key
+   * @method deriveScopedKeys
+   * @param {object} options - required set of options to derive a scoped key
+   * @param {string} options.inputKey - input key that the scoped key is derived from
+   * @param {string} options.keyMaterial - a 32-byte string of additional entropy specific
```

Comment:
Looks like there should be another part of a sentence here after "entropy specific ..."?

Reply:
👍